compiler/src/dmd/typesem.d

```diff
@@ -3444,8 +3444,16 @@ Expression getProperty(Type t, Scope* scope_, Loc loc, Identifier ident, int fla
                 auto s2 = scope_.search_correct(ident);
                 // UFCS
                 if (s2 && s2.isFuncDeclaration)
-                    errorSupplemental(loc, "did you mean %s `%s`?",
-                        s2.kind(), s2.toChars());
+                {
+                    if (s2.ident == ident)
+                    {
+                        errorSupplemental(s2.loc, "cannot call %s `%s` with UFCS because it is not at module scope",
```

Comment:
"because it is not declared at module scope".

and I know they can just search for it, but are we expecting people to know what UFCS is if they hit this error?

Reply:
I know it can still be confusing to new users, but having long explanations in error messages is not our style currently. I am considering adding a standard way to link to the specification, or another external source with more elaborate explanations of error messages.